### ISSUE TITLE
[V3] TypeError: Cannot convert undefined value to object 

### ISSUE THREAD

**Issue by vanenshi** [software-mansion/react-native-reanimated#4139]
### Description

when I am passing an array of styles in `Animated.View` and one of them is undefined, the `hasInlineStyles` function throw error.
The only thing I did was update the library to version 3.0.0

### Steps to reproduce

1. make a simple animate view with an array of styles, and make one of them `undefined` 
```jsx
const styleToOverwrite = undefined

<Animated.View style={[baseStyle, styleToOverwrite]}>
  ...
</Animated.View>
```

### Snack or a link to a repository

expo not suporting the reanimate 3.0.0

### Reanimated version

3.0.0

### React Native version

0.70.7

### Platforms

Android, iOS

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

iOS simulator

### Device model

IPhone 14 

### Acknowledgements

Yes

**Comment by dmahajan980** [comment]
I think this will be fixed after https://github.com/software-mansion/react-native-reanimated/pull/4128 is merged.

**Comment by Jawish-a** [comment]
same issue here, using react navigation.



`ERROR  TypeError: Cannot convert undefined value to object

This error is located at:
    in AnimatedComponent(View)
    in Unknown (created by Drawer)
    in RCTView (created by View)
    in View (created by AnimatedComponent(View))
    in AnimatedComponent(View)
    in Unknown (created by PanGestureHandler)
    in PanGestureHandler (created by Drawer)
    in Drawer (created by DrawerViewBase)
    in DrawerViewBase (created by DrawerView)
    in RCTView (created by View)
    in View (created by GestureHandlerRootView)
    in GestureHandlerRootView (created by DrawerView)
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider (created by SafeAreaInsetsContext)
    in SafeAreaProviderCompat (created by DrawerView)
    in DrawerView (created by DrawerNavigator)
    in PreventRemoveProvider (created by NavigationContent)
    in NavigationContent
    in Unknown (created by DrawerNavigator)
    in DrawerNavigator (created by MainStack)
    in MainStack (created by Main)
    in Main (created by App)
    in Suspense (created by App)
    in QueryClientProvider (created by App)
    in EnsureSingleNavigator
    in BaseNavigationContainer
    in ThemeProvider
    in NavigationContainerInner (created by App)
    in I18nextProvider (created by App)
    in App
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in AwesomeTSProject(RootComponent), js engine: hermes`

packages;

`        "@react-navigation/bottom-tabs": "^6.5.7",
        "@react-navigation/drawer": "^6.6.2",
        "@react-navigation/native": "^6.1.6",
        "@react-navigation/native-stack": "^6.9.12",
        "@react-navigation/stack": "^6.3.16",`

**Comment by MohamedMegdich1** [comment]
Same issue here

**Comment by rush86999** [comment]
same issue

**Comment by tomekzaw** [comment]
Fixed with #4159

**Comment by Mehmet-Alemdar** [comment]
> Same problem
> 
> **App.js**
> 
> ```
> import { NavigationContainer } from '@react-navigation/native'
> import { createDrawerNavigator } from '@react-navigation/drawer';
> 
> import Home from './src/screens/home';
> 
> const Drawer = createDrawerNavigator();
> 
> const App = () => {
>   return (
>     <NavigationContainer>
>       <Drawer.Navigator>
>         <Drawer.Screen name="Home" component={Home} />
>       </Drawer.Navigator>
>     </NavigationContainer>
>   );
> }
> 
> export default App;
> ```
> 
> **home.js**
> 
> ```
> import { View, Text } from 'react-native';
> 
> const Home = () => {
>   return(
>     <View>
>       <Text>Home</Text>
>     </View>
>   )
> };
> 
> export default Home;
> ```
> 
> **error**
> 
> ```
>  ERROR  TypeError: Cannot convert undefined value to object
> 
> This error is located at:
>     in AnimatedComponent(View)
>     in Unknown (created by Drawer)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent(View))
>     in AnimatedComponent(View)
>     in Unknown (created by PanGestureHandler)
>     in PanGestureHandler (created by Drawer)
>     in Drawer (created by DrawerViewBase)
>     in DrawerViewBase (created by DrawerView)
>     in RNGestureHandlerRootView (created by GestureHandlerRootView)
>     in GestureHandlerRootView (created by DrawerView)
>     in RNCSafeAreaProvider (created by SafeAreaProvider)
>     in SafeAreaProvider (created by SafeAreaInsetsContext)
>     in SafeAreaProviderCompat (created by DrawerView)
>     in DrawerView (created by DrawerNavigator)
>     in PreventRemoveProvider (created by NavigationContent)
>     in NavigationContent
>     in Unknown (created by DrawerNavigator)
>     in DrawerNavigator (created by App)
>     in EnsureSingleNavigator
>     in BaseNavigationContainer
>     in ThemeProvider
>     in NavigationContainerInner (created by App)
>     in App
>     in RCTView (created by View)
>     in View (created by AppContainer)
>     in RCTView (created by View)
>     in View (created by AppContainer)
>     in AppContainer
>     in zeitgeist(RootComponent), js engine: hermes
> ```
> 
> **package.json**
> 
> ```
> ...
>   "dependencies": {
>     "@react-navigation/drawer": "^6.6.2",
>     "@react-navigation/native": "^6.1.6",
>     "react": "18.2.0",
>     "react-native": "0.71.3",
>     "react-native-gesture-handler": "^2.9.0",
>     "react-native-reanimated": "^3.0.0",
>     "react-native-safe-area-context": "^4.5.0",
>     "react-native-screens": "^3.20.0",
>     "styled-components": "^5.3.8"
>   },
> ...
> ```

I solved the problem.

firstly I installed the "react-native-reanimated" using yarn

    yarn add react-native-reanimated

then I added "react-native-reanimated/plugin" to the babel.config.js in

    module.exports = {
      presets: [
        ...
      ],
      plugins: [
        ...
        'react-native-reanimated/plugin',
      ],
    };

> don't forget to `pod install` for ios

finally I started the project like this:

for yarn

    yarn start --reset-cache
for npm

    npm start -- --reset-cache

if you're using Expo

    expo start -c


[source of solution][1]


  [1]: https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation/#babel-plugin
